### DISCUSSION TITLE
Successfully replicated on S100101 (Miami) and integrated in FHEM (german)

### DISCUSSION THREAD

**Discussion by Techtobi83** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#559] [General]
**I would like to briefly thank the people responsible for this.** 
A very great project with great instructions, also in detail and a nice community here. I was able to solve all my problems during the construction by researching this part of the plattform. **Thank you again for that.**

Some hints on my part, maybe it will help someone else before some photos below:
- ESP8266 from AZdelivery, V3.x worked fine
- WLAN range was poor, so I soldered on an external antenna (Google: ESP8266 antenna mod)
- Without using low pass filter, DSP and buttons was flickering (solved with capacitors and resistors according to the instructions)
- Integrated into my smart home based on FHEM
- If someone has questions how to integrate this project via MQTT to FHEM, please let me know. 
- My FHEM is connected to Homebridge, so the functions are now also available in Apple Homekit and can be controlled with voice commands via Siri.

Last but not least: 
There are some leftover parts that I am happy to sell on to interested people:
1x ESP8266 flashed and ready to use
2x PCB V2
if anyone is interessted, please feel free to contact me by eMail:  
- edit: all leftover parts sold! -

I have posted some pictures and a link to this github project on my (german) twitter account ( https://twitter.com/tech_tobi/status/1676693224872632321 ) 

Some pictures:
![IMG_4607](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/de14da09-68d1-4a35-af21-b689934a59a7)
![Screenshot 2023-07-06 121438](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/c0e16a50-145a-4568-a750-ee8244dcb454)
![82fcf228-c274-46ff-83db-ce84fec89e37](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/60417b31-f7c0-4b5d-98a8-23db0ebf5a05)
![F0TPjK7WwAIu_eg](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/ee08cf95-48ad-4caf-901f-50ca08dd56de)
![F0TPjLDXwAAZi6N](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/3441ffcd-3ede-4a36-afe1-6f1e6fc686e4)

**Comment by Techtobi83** [comment]
Short instructions for integration in FHEM using MQTT:

- Activate MQTT on the ESP32 in the menu and enter the IP address including port (1883) of the FHEM instance
- Also enter the password and username of the FHEM user

1.) MQTT Server 
You need an standalone MQTT-Server.
These instructions assume that none exist yet and that the server integrated in FHEM is to be used
"define MQTT2_FHEM_Server MQTT2_SERVER 1883 global"
In this new FHEM-device please activate attribute "autocreate" to "simple"
Now you have a running MQTT-Server

2.) Integrate LayZ-Spa Module in FHEM
If everything is correct, the FHEM-device of your layZ-ESP3266 should be recognized and created automatically as soon as the first attempt to send was recognized by your MQTT-Server.  The new device contains a large number of readings and information that are automatically updated according to the specified interval. 
The assignment to the designations is stored in the tutorial by VisualApproach. (Reading AIR = 1 means: bubbles activated, TGT = target temperature... and so on)

3.) Optimizing the FHEM-device
In order not only to receive data, but also to send it conveniently from FHEM, the following attributes are necessary:

setStateListe: on off

stateFormat: TMP

setList: 
pump:on,off { my $on = $EVTPART1 eq 'on'?1:0;;;; qq(layzspa/command  { "CMD":4, "VALUE":$on, "XTIME":0, "INTERVAL":0 }) }
air:on,off { my $on = $EVTPART1 eq 'on'?1:0;;;; qq(layzspa/command { "CMD":2, "VALUE":$on, "XTIME":0, "INTERVAL":0 }) }
unit:Celsius,Fahrenheit { my $Celsius = $EVTPART1 eq 'Celsius'?1:0;;;; qq(layzspa/command  { "CMD":1, "VALUE":$Celsius, "XTIME":0, "INTERVAL":0 }) }
heater:on,off { my $on = $EVTPART1 eq 'on'?1:0;;;; qq(layzspa/command  { "CMD":3, "VALUE":$on, "XTIME":0, "INTERVAL":0 }) }
TGT:slider,20,1,40 layzspa/command  {CMD:0,VALUE:$EVTPART1,XTIME:0,INTERVAL:0}

If all work is done, header of your FHEM-Device looks like this and you are ready to use in FHEM:
![Screenshot 2023-07-06 124928](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/821d6392-722f-40b2-93ac-9d8e20b3d249)

Note: I didn't invent the content myself either, but used the German FHEM forum. The thread to read is here: https://forum.fhem.de/index.php?topic=128077.0

**Comment by AndyMu97** [comment]
Congrats to you for getting it running... I'm wondering why my setup is not working as I thought it is because of the blue LLC and wanted to switch over now to the red LLC. Did you do any changes on the config?

**Reply by Techtobi83** [reply]
no i didn´t changed anything. In "hardware setup" i chose 2x 6-wire 2021 and PCB V2N (whatever it was called).
During soldering i used solution with crossed cables for the connector-cables, thats it. 

I used the blue LLC only, cause my order for 1 red LLC is still missing (didn't want to wait any longer).

**Reply by AndyMu97** [reply]
What do you mean with "crossed cables"?

**Reply by AndyMu97** [reply]
Got it... Page 9 on the instructions PDF. Haven't seen this... need to give it a try first before trying it with red LLC

**Reply by AndyMu97** [reply]
It is now working with the new cabling! Just only one issue:
After turning on the spa, I need to open the pump case and plug in the connector to the pump. Nodemcu is coming up, then connect the display connector. After that the display is working, all fine except the "sound", but this is something known.

But... when I disconnect the spa from power and repower it, the nodemcu will not come up again!
@Techtobi83 Do you also have this issue?
I do have the Lolin nodemcu v3 with the blue LLC, but cables are on H1A and H2A

**Reply by Techtobi83** [reply]
No. When i power off the hole pump (means: get the plug out of the socket) and repower again, esp32 works fine after some seconds. 

I recommend using low pass filter by soldering capacitors and resistors (and then re-soldering the connection cables to the other ports

**Reply by AndyMu97** [reply]
Well, I did this now, too. Same behaviour.

**Reply by Alexxx2005** [reply]
Hallo Tobi, muss diese eine Leitung gekreuzt sein ? Ich habe genau dieselbe Pumpe aber weder mit dem roten noch mit dem Blauen LLC funktioniert es. Ich vermute das dies nun mein Problem sein kann. Grüße Alex

**Reply by Botap** [reply]
Hi, I have an Off Topic question but I really like the wooden cover around your pump. Do you have a build instruction for that?

**Comment by Techtobi83** [comment]
Hi everybody, attached some pictures about PCBs i have to sell (located in germany). Feel free to contact me by mail techtobi(at)planetdraco(dot)de or here on github or twitter @Tech_tobi if you are interessted. 

1x PCB V2 with completely flashed ESP3266 & red LLC
2x PCB V2 "nacked" 
![tempImageN0OPJm](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/7d9de766-3b49-4ad5-a195-2d59a8168c37)

![tempImageZTUoDt](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/942ca066-852c-4ffc-916d-f78eb4655105)

![tempImagelTzzHW](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/86073272/de6877f4-b561-4c39-95d1-6f19d0d67070)

**Reply by Techtobi83** [reply]
In the meantime, all my individual parts have been sold. Please do not ask anymore, sorry.

**Reply by reevery** [reply]
@Techtobi83 Could you edit your previous posts advertising parts for sale please? Then people don't read, think you have some, send you a reply, then read your comment that you sold them all? Thank you

**Reply by Techtobi83** [reply]
Done

**Comment by bartbakels** [comment]
Still pcbs for sale, my selfcreated one encoubters some issues lately

**Reply by Techtobi83** [reply]
i still have 2 boards for sale, yes.

**Reply by bartbakels** [reply]
Willing to send them to me 1 or 2 and are they including the components ex esp? Possible we can mail on how to arrange this; bartbakels@ziggo.nl
